test/functional/pdo_sqlsrv/pdo_ae_output_param_char_size.phpt

```diff
@@ -0,0 +1,173 @@
+                dropProc($conn, $spname);
+                $spSql = "CREATE PROCEDURE $spname (
+                                @c_det $type OUTPUT, @c_rand $type OUTPUT ) AS
+                                SELECT @c_det = c_det, @c_rand = c_rand FROM $tbname";
+                $conn->query($spSql);
+                
+                // call stored procedure
+                $outSql = getCallProcSqlPlaceholders($spname, 2);
+                foreach ($pdoParamTypes as $pdoParamType) {
+                    $det = "";
+                    $rand = "";
+                    $stmt = $conn->prepare($outSql);
+                    trace("\nParam $pdoParamType with INOUT = $inout\n");
+                    
+                    if ($inout) {
+                        $paramType = $pdoParamType | PDO::PARAM_INPUT_OUTPUT;
+                    } else {
+                        $paramType = $pdoParamType;
+                    }
+                    
+                    $len = $length;
+                    if ($pdoParamType == PDO::PARAM_BOOL || $pdoParamType == PDO::PARAM_INT) {
+                        $len = PDO::SQLSRV_PARAM_OUT_DEFAULT_SIZE;
+                        $det = $rand = 0;
+                    }
+                    
+                    $stmt->bindParam(1, $det, $paramType, $len); 
+                    $stmt->bindParam(2, $rand, $paramType, $len); 
+                    
+                    try {
+                        $stmt->execute();
+                        $errMsg = "****$type as $pdoParamType failed with INOUT = $inout:****\n";
+                        if ($length < 64 && $pdoParamType != PDO::PARAM_STR) {
+                            if ($pdoParamType == PDO::PARAM_BOOL) {
+                                // For boolean values, they should all be bool(true)
+                                // because all "string literals" are non-zeroes
+                                if (!$det || !$rand) {
+                                    printValues($errMsg, $det, $rand, $input0, $input1);
+                                }
+                            } else {
+                                // $pdoParamType = PDO::PARAM_INT
+                                // Expect numeric values
+                                if ($det != intval($input0) || $rand != intval($input1)) {
+                                    printValues($errMsg, $det, $rand, $input0, $input1);
+                                }
+                            }
+                        } elseif ($det !== $input0 || $rand !== $input1) {
+                            printValues($errMsg, $det, $rand, $input0, $input1);
+                        }
+                    } catch (PDOException $e) {
+                        $message = $e->getMessage();
+                        $errMsg = "EXCEPTION: ****$type as $pdoParamType failed with INOUT = $inout:****\n";
+                        if ($pdoParamType == PDO::PARAM_NULL || $pdoParamType == PDO::PARAM_LOB) {

```

Comment:
I'm surprise PARAM_LOB doesn't work. I always thought the driver treat PARAM_STR and PARAM_LOB the same

Reply:
No they are not treated the same when binding as output parameters